### ISSUE TITLE
Code not up to date

### ISSUE THREAD

**Issue by rancidfrog** [Slion/Fulguris#178]
Latest release seems to be 1.4.8:
![](https://i.imgur.com/8DdyUQ6.png)

Yet,
![](https://i.imgur.com/NUavACv.png)

Also, I was a little confused as releases section shows 5.1.0, which is Lightning releases.
You should clear releases section and create your own Fulguris tags, hopefully with apks included.
I believe this can be done automatically either per Tag creation or even per commit.

**Comment by Slion** [comment]
Oups, yeah still need to push the latest changes.

> Also, I was a little confused as releases section shows 5.1.0, which is Lightning releases.

Interesting, thought I disabled that since releases are published on slions.net, however the URL still works.
I'm not sure how to clear that section, would need to look into it.

**Comment by Slion** [comment]
Just pushed the latest changes. I also deleted the legacy Lightning releases from https://github.com/Slion/Fulguris/releases
@rancidfrog Did you want to contribute something?

**Comment by jamal2362** [comment]
Hi,
Would it be possible to push the latest changes ?

I find some things interesting that could be implemented in Styx. :)

**Comment by Slion** [comment]
Just pushed the latest changes.

**Comment by jamal2362** [comment]
bump 😁

**Comment by Slion** [comment]
> bump 😁

We are in the process of reviewing Fulguris licensing to avoid rebranding efforts taking advantage of Fulguris contributors without proper attribution. While this process is on going there could delays to source code delivery.
See: https://github.com/jamal2362/Styx/issues/75

**Comment by jamal2362** [comment]
@Slion 

As you already noticed, the credit is available in the app as well as on GitHub. Don't understand where the problem should be?

**Comment by Slion** [comment]
As earlier discussed in various other places on the Internet, problems with Styx are as fellow:
- Code is copied from Fulguris and pasted in Styx code base, in the process actual author name is replaced with your name.
- No effort is made of your part to contribute back to the Fulguris code base.
- Over the months were you kept feeding on that code base you made no effort to promote Fulguris beyond the legally binding.

To me that situation is not sustainable. Therefore it is likely that some Fulguris features may remain closed source for the time being to avoid such plunder.

**Comment by IzzySoft** [comment]
@jamal2362 this situation was brought to my attention. As you've closed your issue tracker and made existing issues unavailable, I can neither contact you there nor check those references. But the situation as described contradicts the spirit of FOSS. What notice should I apply to your app's listing in my repo? That it violates the very license both apps use, namely 3.4 of the MPL-2.0, which reads:

> You may not remove or alter the substance of any license notices (including copyright notices, patent notices, disclaimers of warranty, or limitations of liability) contained within the Source Code Form of the Covered Software, except that You may alter any license notices to the extent required to remedy known factual inaccuracies.

It doesn't do taking commits from one developer and declaring them your own, claiming you wrote and added that code – even if that paragraph doesn't explicitly mention commits. Wouldn't that mean you do not abide to the terms, and thus your declaration is invalid? Must I then remove your app from my repo, and should a complaint filed to Play Store for the very same reasons? Before drawing conclusions, I'd like to hear what you have to say – though I have not much time for "negotiations".

As the saying goes "never attribute to malice that which is adequately explained by stupidity" (Hanlon's razor), let me assume you simply chose the wrong technique to integrate upstream changes into your app. But now that you're aware of it, what will you do about it?

@Slion apologies for hijacking your issue for that, but with the Stux issue tracker being closed and the context so conveniently available here, I hope that's acceptable for you.

**Comment by jamal2362** [comment]
Hey @IzzySoft 

The credits can be found in the app under About. There are all the browsers linked, of which I have used the code.

If you like, feel free to take the app out of your repo. The reason I turned off the issue tracker is because I'm not currently working on the browser. I have removed the Izzy banner that leads to your repo.

Greetings,
Jamal ✌️

**Comment by IzzySoft** [comment]
Hi @jamal2362 – yes, I've read you put the credits into the About screen. That's why I assume "bad choice of tools" (or incomplete understanding) rather than "bad intent". I was hoping you'd see why commit rewriting is a bad idea and should not be done (as it removes attribution from the original committer, and makes it real hard to find out who contributed what code – there's a reason why the committer is logged). And you see where that leads to: those authors (rightfully) feel betrayed, FOSS people consider closing their code because of that.

My goal was not removing your app from my repo. But if you choose so **instead of** fixing the cause, the next steps would probably also include reporting your app at Play for plagiarism. That would be a hard choice to make, and should rather be avoided. But again, mentioning credits in the About screen is good and should definitely be done – but it's not making up for commit rewriting. Attribution should not be removed where it exists.

Thanks for reactivating your issue tracker – though I wonder where the linked issues have gone to (e.g. for the above I just get a note it has been deleted – which again leaves a taste; I assume you had good reasons to do so, but one could think you wanted to get rid of something "unfortunate". That a browser like Styx only ever had a single issue opened is hard to believe as well :wink:).

I imagine it would be hard to restore all those past commits to their original authors – but how do you plan to deal with this in the future? I'd love to further promote your app (e.g. by giving it a presence in my repo and in my app listings). Will you give that a chance? Or did you stop development altogether, and have abandoned Styx?

**Comment by jamal2362** [comment]
Hey @IzzySoft 

It's really ok if the app remove from their repo. The app is currently at the end and I have other projects.

I wouldn't care about a Google Play Store notification either. Actually, I don't care who uses Styx. Originally it was just me using the app and not publicly available to everyone.

I will remove the Styx repo today so that Fulguris gets more attention. 

It was never the intention to steal anything or anything else. Styx was just meant to be a mix between Fulguris and SmartCookieWeb.

But anyway. I will only use Styx 15.4 privately for myself and will remove the app from the Play Store and delete the repo.

Greetings,
Jamal

**Comment by jamal2362** [comment]
Hey @IzzySoft 

I would be very happy if they would remove all three apps from their IzzyOnDroid repo. 

They are no longer available on GitHub and in the Google Play Store.

Greetings,
Jamal

**Comment by Slion** [comment]
Wow that went out of control real fast 🤣 Looks like this is going to save me some lawyers fee too...

@IzzySoft May I ask what prompted this sudden move? I mean that thing has been going on for months if not over a year. Had I known you yield some much power I would have contacted you myself long ago.
Any chance of redirecting people from the Styx page to Fulguris with an appropriate explaination?

After taking a break from the project mostly out of disgust about the situation I lately invested quite a bit of time and effort in research concerning relicensing and multiple licensing solutions. I'm contemplating something like CPAL in the hope future rebranding would at least properly credit the upstream project.

**Comment by jamal2362** [comment]
As I wrote before, I'm not interested in this project anymore, so I don't give a shit. And your imaginary lawyer is ridiculous. 😂

**Comment by IzzySoft** [comment]
> It was never the intention to steal anything or anything else.

Thanks – that was my assumption as well (as I hope was visible from my writings). And for "personal use" I see no issues at all with your "commit policy". It's just not good for "public stuff".

> I will remove the Styx repo today so that Fulguris gets more attention.

May I suggest to instead just archive it, after adding a fitting note to its Readme ("EOL, no longer maintained etc")? Deleting could leave the smell of "removing traces", which again is not your intention. I'd then set Styx to `NoSourceSince` in my repo – or remove it should you prefer that.

Not sure what you mean by "remove all three apps". Assuming by the other two you mean Fulguris and SmartCookieWeb, they are not in my repo. You probably mean your other two apps where you removed the Github repos – DeepL and URL-Radio? I've just marked them `NoSourceSince` for now as their repos are gone.

> I no longer want these apps to be offered to the public out of respect for other developers

:bow: A hard consequence, my respect. Though I think the resp. developers would agree to your keeping the apps up if you'd simply not remove attribution from the commits any longer (a note that this was mistakenly done in the past would certainly be appreciated – but the main thing would be to properly integrate upstream commits with their committer attributes). Of course this is your choice. Still, let me point out once more my goal was not to seal away your apps, but rather to keep them app in a way all affected devs can live with.

That said: thanks for your cooperation in this issue, Jamal. If you need a helping hand in getting things straight, I'm certainly not the only one willingly offering it (though my time is a bit limited, and I cannot be everywhere – I'll try my best). Would really love to see all these projects living in friendly cooperation!

----

@Slion What was prompting my move was a user asking me to help with this – and I said I had no idea how to, but would leave a note. I had no idea concerning the outcome – but from what I felt saw a good chance of Jamal being well-meaning, and the issue just based on "bad choice of tooling" (which luckily proved to be the case, which I'm glad for).

And I meant what I wrote: ideally, you work together. Whether it's on a single project like Fulguris where Jamal contributes his ideas, or on two separate projects with a slightly different focus where both sides participate each with their contributions (which I'd actually feel better with, and which would also give users something to choose between). I'm allowed to dream – to make such dreams come true is up to the two of you :smile:

But at least I hope you'll now keep your project open, in the spirit of FOSS. That's something I feel important – we need more good FOSS projects :smiley:

----

@jamal2362 let's better drop the lawyer topic, I've seen some of those fights and wish we all can do without. Don't doubt the lawyer's powers even with FOSS licenses – believe me, it exists (I've seen some of it with several other projects); but it's much better to live without such needs, for all of us. Much more peaceful :smile:

**Comment by jamal2362** [comment]
Thank you for your words @IzzySoft 👍

As I said.
I have no more interest in the Styx project so I removed everything to avoid more discussion in the future.

By deleting the 3 apps I meant Styx, URL Radio and DeepL.

In the meantime I have other goals to focus on, so a collaboration with @Slion is not possible.

For me this project or mine is done. Forever.

I am also not angry, sad etc.
I am satisfied currently. 🤗

**Comment by IzzySoft** [comment]
Thanks @jamal2362 – no bad feelings from any side but instead relief is a good outcome I'd say! Guess you'd be OK if I keep Styx in my repo for a little longer, and append the description with some hints (friendly words about EOL of Styx and pointing to Fulguris where development is ongoing and some of Styx' ideas went to, plus SmartCookieWeb as alternative)? Similarly, I'd update the description of your other two apps. If you prefer removal I'd of course comply, though I'd prefer keeping them all at least a little longer, if only for existing users to be notified.

@Slion I hope you'll keep up Fulguris under FOSS terms now? Did you consider making it available at F-Droid or via my repo? I see you've already prepared a productFlavor for it (F-Droid wouldn't accept Firebase or Crashlytics, which are avoided via that flavor). Would be good if I can point to an "open place" where users can obtain the app, not to a "walled garden" like Play Store.

**Comment by Slion** [comment]
> I hope you'll keep up Fulguris under FOSS terms now?
So do I. I'm still considering moving to [CPAL](https://en.wikipedia.org/wiki/Common_Public_Attribution_License) one way or another. Whether you consider such badgeware license FOSS or not is apparently up for debate.

> Did you consider making it available at F-Droid or via my repo?
[F-Droid](https://github.com/Slion/Fulguris/issues/84) yes, but somehow we keep running into obstacles. It's actually harder to get published on F-Droid than it is on Google Play. Who knew! 
Your repo, no, never heard of it TBH until Styx came along 🤷‍♂️I have nothing against it though.

> Would be good if I can point to an "open place" where users can obtain the app
What's wrong with:
http://fulguris.slions.net
Though admittedly the Google Play badge is much easier to spot than the green download button in the top right corner.
Release history is accessibly from the Updates tab and earlier versions can be downloaded from the History tab.

**Comment by IzzySoft** [comment]
> So do I. I'm still considering moving to CPAL one way or another. Whether you consider such badgeware license FOSS or not is apparently up for debate.

No need to debate: As [SPDX shows](https://spdx.org/licenses/), CPAL-1.0 is approved by both OSI and FSF, so it's acceptable even by the strong F-Droid standards.

> It's actually harder to get published on F-Droid than it is on Google Play.

Hehe… certainly depends, but we¹ can be quite … well, strict. Especially when it comes to proprietary or even just binary parts in the source repo. I cannot tell what obstacles you refer to as I found neither RFP nor merge request for Fulguris.

> Your repo, no, never heard of it TBH

And there I'm living in my ivory tower day-dreaming of high positions :rofl: Never mind: it's quite wide spread meanwhile (just use the search engine of your least mistrust and look for "IzzyOnDroid repo") – but considering there are people who didn't even hear about F-Droid itself, I should be meek and humble :smile:

> I have nothing against it though.

Glad to read! I'd just need a way to get hold of the APKs then. The easiest way is finding them attached to their corresponding Github releases. My updater then can automatically take care and grabs new APKs within 24h of your making them available. If that's not an option for you, a "fixed URL" (changing file-names won't do for automated update checks as my updater cannot predict them) returning a proper LAST_MODIFIED header would work as well.

> What's wrong with: http://fulguris.slions.net

Well, one part you already self-answered – the other I mentioned, a third would be "insecure URL" (no https – though that's fixed easily by just adding the `s` in the proper place :laughing:). Further, in a proper repo, updates are taken care for via the associated app (here: one of the [available F-Droid clients](https://android.izzysoft.de/applists/category/named/apps_markets#group_1181)). That's usually also what users head to when looking for apps they might need – so being present there makes it easier to be found. And as you speak about badges: [take a look here](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) :smiley:

Apart from that: A nice page indeed! I'd definitely point to that as "homepage" from the app description (there are defined fields for several locations; be welcome to use [my repo browser](https://apt.izzysoft.de/fdroid) to check out some examples from different repositories, or check with [F-Droid.org](https://f-droid.org/) for what's shown there).

----

<sup>¹ disclosure: I'm one of the maintainers there</sup>

**Comment by Slion** [comment]
> I'm one of the maintainers there

I figured 😁

>  I cannot tell what obstacles you refer to as I found neither RFP nor merge request for Fulguris.

https://gitlab.com/fdroid/rfp/-/issues/1442
I just scanned that issue again and it looks like it was left unattended after some bot published a report containing errors.
Looks like you had concerns about your tool chain not supporting unicode characters in the description and android app id prefix.

It obviously not the best time for me to work on that as I still need to tackle those license issues and how to open source my latest contributions first but feel free to pursue this topic on GitLab or on #84.
